src/file-input/file-dropzone.svelte

```diff
@@ -0,0 +1,125 @@
+<script>
+  import Paperclip from './paperclip.svelte';
+  import Plus from './plus.svelte';
+  import FileTile from './file-tile.svelte';
+  import { createEventDispatcher } from 'svelte';
+  import classes from '../_utils/classes.js';
+  import ripple from '../_utils/ripple.js';
+
+  let _class = null;
+  export { _class as class };
+
+  export let value = [];
+  export let fileComponent = FileTile;
+  export let accept = null;
+
+  let dragActive = false;
+
+  let emptyLayer;
+  let dropzoneLayer;
+  let inputElement;
+
+  function mimeMatches(acceptedType, acceptedSubtype, fileType, fileSubtype) {
+    if (acceptedType == null) {
+      return true;
+    }
+
+    if (fileType == null) {
+      return false;
+    }
+
+    if (acceptedType !== fileType) {
+      return false;
+    }
+
+    if (acceptedSubtype === '*') {
+      return true;
+    }
+
+    return acceptedSubtype === fileSubtype;
+  }
```

Comment:
The input won't, but this is for drag'n'drop, which isn't even connected to that input

Reply:
Right. I thought if we connect them together using `bind:files` that the `<input>` will validate the types first, but it seems like programmatically adding files to it skips the check.

Reply:
However, the check is still flawed. There are 2 cases that need to be handled:
- `accept` can be a comma separated list (with possible whitespace) of types to accept
- `accept` can specify file extension instead of MIME type (ex: `.pdf`).

That can also create some inconsistency depending on the file type. For example. the user might set `accept=".pdf"`, while `file.type` would be "application/pdf". The user can also choose some extensions to accept, but the browser might not detect a type for them and would have `file.type === ''`.

Reply:
Oh, okay, I see. Then I need to level up that check

Reply:
case-sensitivity is also something to be considered